### PR TITLE
Fix #4460: Add Wiki documentation for adding the spotlight experience to any element 

### LINKED ISSUE #4460: Add Wiki documentation for adding the spotlight experience to any element

### PR DESCRIPTION
**Pull request by JishnuGoyal**
## Explanation

Fixes #4460 
This PR adds a new page to the Wiki explaining how Spotlights should be used, how new ones should be added and how should they be tested.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by JishnuGoyal** [comment]
Additionally, here is this link to the original Spotlight Wiki doc: https://docs.google.com/document/d/1houcOoZLbTgdwnL1HnmocolEjP6exfY-R_qsNbYPY7Y/edit#

File: wiki/Spotlight Guide
Comment (BenHenning) [inline wiki/Spotlight Guide:1]: This file needs to end with '.md'.

**Review by BenHenning — COMMENTED** [review]
Thanks @JishnuGoyal! I think the filename is incorrect. Also we need to add this to the _Sidebar.md file so that it can be found on the side.

Besides that, since I've already reviewed the contents of this file in detail, I'm actually going to defer to @seanlip for the PR review since he's also broadly going to be reviewing wiki pages moving forward.

**Comment by BenHenning** [comment]
@JishnuGoyal please assign to @seanlip once the comments above are addressed.

File: wiki/Spotlight Guide
Comment (JishnuGoyal) [reply wiki/Spotlight Guide:1]: done

**Comment by JishnuGoyal** [comment]
Hey @BenHenning, how exactly should I make sure this file is shown in the sidebar?

**Comment by JishnuGoyal** [comment]
PTAL @seanlip , thanks!

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: At the start: "Spotlights are a visual tool that ..."

Maybe, below this paragraph, add a new one that's like: "Here is an example of a spotlight in the Home screen:" (or wherever) and add a picture to show what a spotlight looks like in practice.

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: This isn't rendering properly, you may need to put `- ` at the start of this and the next line to get bullet points to show up in markdown.

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: Ditto for these fields. Please use the markdown preview functionality in GitHub to check your work.

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: one of --> either

End with a period.

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: Same thing re bullet point formatting.

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: method --> methods

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: Deindent this by 2 perhaps? This should be in a separate indentation block from the next 4 lines.

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: Indent this (by whatever is needed to match the corresponding '(' above)

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: Make sure the indentation of this matches checkNotNull() above

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: This results in the reader asking the question: "Which platform parameter value?"

Either make the next couple of points bullet points, or make them separate paragraphs and use bold highlighting for the initial phrase (e.g. **In tests:** ..., **In production:** ...).

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: Should this be a new paragraph? If so, add a newline above this one. Please check and do a self-review using the Markdown preview functionality on GitHub.

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @JishnuGoyal! Took a full pass, PTAL.

Re including this page in the sidebar, @BenHenning's already given you a pointer -- please add it to the relevant place in `_Sidebar.md`.   Thanks!

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

File: wiki/Spotlight Guide.md
Comment (seanlip) [inline wiki/Spotlight Guide.md]: either --> be either

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: Done - I'm wondering if there's a way to make this image look smaller?

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done, PTAL

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done, PTAL

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done, PTAL

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done, PTAL

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

File: wiki/Spotlight Guide.md
Comment (JishnuGoyal) [reply wiki/Spotlight Guide.md]: done

**Comment by JishnuGoyal** [comment]
Please take another pass at this @seanlip , thanks!

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @JishnuGoyal, I think the page content looks fine. But I think you missed @BenHenning's comment from before. Could you please:

(a) Add this page to the _Sidebar.md file so that users can get to it
(b) Make sure this page is named in the same way as others in the wiki (no spaces in the name, use hyphens)

Once you've done that, this PR can be merged. Thanks!

**Comment by JishnuGoyal** [comment]
Updated the file name and added the same to the sidebar. 

@seanlip PTAL, thanks!

**Review by seanlip — APPROVED** [review]
Thanks! LGTM.

File: wiki/Spotlight Guide.md
Comment (seanlip) [reply wiki/Spotlight Guide.md]: Probably the easiest is to use a smaller image? Feel free to do that in a separate PR though.

**Comment by JishnuGoyal** [comment]
I have updated this branch. @BenHenning PTAL, thanks!

**Comment by seanlip** [comment]
Merged. Thanks @JishnuGoyal!

**Pull request merged**
